ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/manager/task/GetHeadersFromPeerByHashTaskTest.java

```diff
@@ -97,9 +113,100 @@ public void getHeadersFromHashWithSkipReversed() {
     getHeadersFromHash(2, true);
   }
 
+  @Test
+  public void completesWhenPeersSendEmptyResponsesAndReducesReputation() {
+    // Setup a unresponsive peer
+    final Responder responder = RespondingEthPeer.emptyResponder();
+    final RespondingEthPeer respondingEthPeer =
+        EthProtocolManagerTestUtil.createPeer(ethProtocolManager, 1000);
+
+    // Setup data to be requested
+    final List<BlockHeader> requestedData = generateDataToBeRequested();
+
+    // Execute task and wait for response
+    final EthTask<AbstractPeerTask.PeerTaskResult<List<BlockHeader>>> task =
+        createTask(requestedData);
+    final CompletableFuture<AbstractPeerTask.PeerTaskResult<List<BlockHeader>>> future = task.run();
+    respondingEthPeer.respond(responder);
+    assertThat(future.isDone()).isTrue();
+    assertThat(future.isCompletedExceptionally()).isFalse();
+    assertThat(respondingEthPeer.getEthPeer().getReputation().getScore()).isEqualTo(99);
+  }
+
+  private List<BlockHeader> dummyBHList(final int count) {
```

Comment:
nit: I think generally we have helper methods at the bottom, after all the `@Tests`

Reply:
done